diff --git i/libs/vkd3d/device_vkd3d_ext.c w/libs/vkd3d/device_vkd3d_ext.c
index 94818b9a..c0057e17 100644
--- i/libs/vkd3d/device_vkd3d_ext.c
+++ w/libs/vkd3d/device_vkd3d_ext.c
@@ -1179,6 +1179,70 @@ CONST_VTBL struct ID3D12DXVKInteropDevice3Vtbl d3d12_dxvk_interop_device_vtbl =
     d3d12_dxvk_interop_device_GetVulkanHeapInfo,
 };
 
+void d3d12_device_notify_vk_swapchain_creation(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
+{
+    if (!device->vk_info.NV_low_latency2)
+        return;
+
+    spinlock_acquire(&device->low_latency_swapchain_spinlock);
+
+    if (device->swapchain_info.low_latency_swapchain && device->swapchain_info.low_latency_swapchain != chain)
+    {
+        /* Hard evidence of multiple swapchains being in flight. */
+        dxgi_vk_swap_chain_set_latency_sleep_mode(chain, false, false, 0);
+        dxgi_vk_swap_chain_decref(device->swapchain_info.low_latency_swapchain);
+        WARN("Multiple swapchains are in-flight. LL2 will be disabled until the situation stabilizes.\n");
+        device->swapchain_info.low_latency_swapchain = NULL;
+    }
+
+    spinlock_release(&device->low_latency_swapchain_spinlock);
+}
+
+void d3d12_device_register_low_latency_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
+{
+    if (!device->vk_info.NV_low_latency2)
+        return;
+
+    spinlock_acquire(&device->low_latency_swapchain_spinlock);
+
+    device->swapchain_info.vk_swapchain_count++;
+
+    if (device->swapchain_info.vk_swapchain_count == 1 && !device->swapchain_info.low_latency_swapchain)
+    {
+        dxgi_vk_swap_chain_incref(chain);
+        device->swapchain_info.low_latency_swapchain = chain;
+        dxgi_vk_swap_chain_set_latency_sleep_mode(chain, device->swapchain_info.mode,
+                device->swapchain_info.boost, device->swapchain_info.minimum_us);
+    }
+
+    /* Defer the demotion of the existing low-latency swapchain.
+     * Supposedly, there are applications that create a second swapchain, but never actually present anything to it
+     * before destroying it.
+     * It will never create a Vulkan swapchain and will not conflict with LL2.
+     * We can defer the demotion until we have evidence later. */
+
+    spinlock_release(&device->low_latency_swapchain_spinlock);
+}
+
+void d3d12_device_remove_low_latency_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
+{
+    if (!device->vk_info.NV_low_latency2)
+        return;
+
+    spinlock_acquire(&device->low_latency_swapchain_spinlock);
+
+    assert(device->swapchain_info.vk_swapchain_count);
+    device->swapchain_info.vk_swapchain_count--;
+
+    if (device->swapchain_info.low_latency_swapchain == chain)
+    {
+        dxgi_vk_swap_chain_decref(chain);
+        device->swapchain_info.low_latency_swapchain = NULL;
+    }
+
+    spinlock_release(&device->low_latency_swapchain_spinlock);
+}
+
 static inline struct d3d12_device *d3d12_device_from_ID3DLowLatencyDevice(d3d_low_latency_device_iface *iface)
 {
     return CONTAINING_RECORD(iface, struct d3d12_device, ID3DLowLatencyDevice_iface);
diff --git i/libs/vkd3d/swapchain.c w/libs/vkd3d/swapchain.c
index 5f2e9355..7140f6c8 100644
--- i/libs/vkd3d/swapchain.c
+++ w/libs/vkd3d/swapchain.c
@@ -623,7 +623,7 @@ static ULONG STDMETHODCALLTYPE dxgi_vk_swap_chain_AddRef(IDXGIVkSwapChain2 *ifac
     {
         dxgi_vk_swap_chain_incref(chain);
         ID3D12CommandQueue_AddRef(&chain->queue->ID3D12CommandQueue_iface);
-        d3d12_device_register_swapchain(chain->queue->device, chain);
+        d3d12_device_register_low_latency_swapchain(chain->queue->device, chain);
     }
 
     return refcount;
@@ -644,10 +644,7 @@ static ULONG STDMETHODCALLTYPE dxgi_vk_swap_chain_Release(IDXGIVkSwapChain2 *ifa
         /* Calling this from the submission thread will result in a deadlock, so
          * drain the swapchain queue now. */
         dxgi_vk_swap_chain_drain_queue(chain);
-
-        if (device->vk_info.NV_low_latency2)
-            d3d12_device_remove_swapchain(device, chain);
-
+        d3d12_device_remove_low_latency_swapchain(device, chain);
         dxgi_vk_swap_chain_decref(chain);
         ID3D12CommandQueue_Release(&queue->ID3D12CommandQueue_iface);
     }
@@ -1714,13 +1711,7 @@ static void dxgi_vk_swap_chain_destroy_swapchain_in_present_task(struct dxgi_vk_
     chain->present.current_backbuffer_index = UINT32_MAX;
 
     if (chain->queue->device->vk_info.NV_low_latency2)
-    {
-        spinlock_acquire(&chain->queue->device->low_latency_swapchain_spinlock);
-        chain->queue->device->swapchain_info.vk_swapchain_count--;
-        spinlock_release(&chain->queue->device->low_latency_swapchain_spinlock);
-
         pthread_mutex_unlock(&chain->present.low_latency_swapchain_lock);
-    }
 }
 
 static VkColorSpaceKHR convert_color_space(DXGI_COLOR_SPACE_TYPE dxgi_color_space)
@@ -2100,6 +2091,10 @@ static void dxgi_vk_swap_chain_recreate_swapchain_in_present_task(struct dxgi_vk
     if (chain->present.is_surface_lost)
         return;
 
+    /* Hard evidence that we're going ahead with swapchain creation, demote any existing LL2 chain
+     * that is not ourselves. */
+    d3d12_device_notify_vk_swapchain_creation(chain->queue->device, chain);
+
     /* If we fail to query formats we are hosed, treat it as a SURFACE_LOST scenario. */
     pthread_mutex_lock(&chain->properties.lock);
     /* This is only called on an event where we have to recreate the swapchain,
@@ -2283,18 +2278,6 @@ static void dxgi_vk_swap_chain_recreate_swapchain_in_present_task(struct dxgi_vk
     /* If low latency is supported restore the current low latency state now */
     if (chain->queue->device->vk_info.NV_low_latency2)
     {
-        struct d3d12_device *device = chain->queue->device;
-
-        spinlock_acquire(&device->low_latency_swapchain_spinlock);
-        device->swapchain_info.vk_swapchain_count++;
-
-        if (device->swapchain_info.vk_swapchain_count > 1 && device->swapchain_info.low_latency_swapchain)
-        {
-            dxgi_vk_swap_chain_decref(device->swapchain_info.low_latency_swapchain);
-            device->swapchain_info.low_latency_swapchain = NULL;
-        }
-        spinlock_release(&device->low_latency_swapchain_spinlock);
-
         dxgi_vk_swap_chain_set_low_latency_state(chain, &chain->present.low_latency_state);
         pthread_mutex_unlock(&chain->present.low_latency_swapchain_lock);
     }
@@ -3958,8 +3941,7 @@ static HRESULT STDMETHODCALLTYPE dxgi_vk_swap_chain_factory_CreateSwapChain(IDXG
         return hr;
     }
 
-    if (chain->queue->device->vk_info.NV_low_latency2)
-        d3d12_device_register_swapchain(chain->queue->device, chain);
+    d3d12_device_register_low_latency_swapchain(chain->queue->device, chain);
 
     *ppSwapchain = (IDXGIVkSwapChain*)&chain->IDXGIVkSwapChain_iface;
     return S_OK;
diff --git i/libs/vkd3d/vkd3d_private.h w/libs/vkd3d/vkd3d_private.h
index 5f94cfcf..b485d75c 100644
--- i/libs/vkd3d/vkd3d_private.h
+++ w/libs/vkd3d/vkd3d_private.h
@@ -6189,33 +6189,9 @@ static inline bool d3d12_device_prefers_render_pass_resolves(const struct d3d12_
             device->workarounds.tiler_suspend_resume_relax_load_store_op;
 }
 
-static inline void d3d12_device_register_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
-{
-    spinlock_acquire(&device->low_latency_swapchain_spinlock);
-
-    if (!device->swapchain_info.low_latency_swapchain)
-    {
-        dxgi_vk_swap_chain_incref(chain);
-        device->swapchain_info.low_latency_swapchain = chain;
-        dxgi_vk_swap_chain_set_latency_sleep_mode(chain, device->swapchain_info.mode,
-                device->swapchain_info.boost, device->swapchain_info.minimum_us);
-    }
-
-    spinlock_release(&device->low_latency_swapchain_spinlock);
-}
-
-static inline void d3d12_device_remove_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
-{
-    spinlock_acquire(&device->low_latency_swapchain_spinlock);
-
-    if (device->swapchain_info.low_latency_swapchain == chain)
-    {
-        dxgi_vk_swap_chain_decref(chain);
-        device->swapchain_info.low_latency_swapchain = NULL;
-    }
-
-    spinlock_release(&device->low_latency_swapchain_spinlock);
-}
+void d3d12_device_register_low_latency_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain);
+void d3d12_device_remove_low_latency_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain);
+void d3d12_device_notify_vk_swapchain_creation(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain);
 
 /* ID3DBlob */
 struct d3d_blob
diff --git a/libs/vkd3d/command.c b/libs/vkd3d/command.c
--- a/libs/vkd3d/command.c
+++ b/libs/vkd3d/command.c
@@ -9837,7 +9837,7 @@ static bool vk_image_copy_from_d3d12(VkImageCopy2 *image_copy,
 
 static void d3d12_command_list_transition_image_layout_with_global_memory_barrier(struct d3d12_command_list *list,
         struct d3d12_command_list_barrier_batch *batch,
-        VkImage vk_image, const VkImageSubresourceLayers *vk_subresource,
+        struct d3d12_resource *resource, const VkImageSubresourceLayers *vk_subresource,
         VkPipelineStageFlags2 src_stages, VkAccessFlags2 src_access, VkImageLayout old_layout,
         VkPipelineStageFlags2 dst_stages, VkAccessFlags2 dst_access, VkImageLayout new_layout,
         VkAccessFlags2 global_src_access, VkAccessFlags2 global_dst_access)
@@ -9855,9 +9855,14 @@ static void d3d12_command_list_transition_image_layout_with_global_memory_barrie
     vk_barrier.newLayout = new_layout;
     vk_barrier.srcQueueFamilyIndex = VK_QUEUE_FAMILY_IGNORED;
     vk_barrier.dstQueueFamilyIndex = VK_QUEUE_FAMILY_IGNORED;
-    vk_barrier.image = vk_image;
+    vk_barrier.image = resource->res.vk_image;
     vk_barrier.subresourceRange = vk_subresource_range_from_layers(vk_subresource);
 
+    /* maint9 quirk. If enabled, and the 3D image is 2D_ARRAY compatible,
+     * layerCount is in terms of slices, so the compatible thing to do is REMAINING_ARRAY_LAYERS. */
+    if (resource->desc.Dimension == D3D12_RESOURCE_DIMENSION_TEXTURE3D)
+        vk_barrier.subresourceRange.layerCount = VK_REMAINING_ARRAY_LAYERS;
+
     need_global_barrier = global_src_access || global_dst_access;
 
     if (need_global_barrier)
@@ -9871,12 +9876,12 @@ static void d3d12_command_list_transition_image_layout_with_global_memory_barrie
 
 static void d3d12_command_list_transition_image_layout(struct d3d12_command_list *list,
         struct d3d12_command_list_barrier_batch *batch,
-        VkImage vk_image, const VkImageSubresourceLayers *vk_subresource,
+        struct d3d12_resource *resource, const VkImageSubresourceLayers *vk_subresource,
         VkPipelineStageFlags2 src_stages, VkAccessFlags2 src_access, VkImageLayout old_layout,
         VkPipelineStageFlags2 dst_stages, VkAccessFlags2 dst_access, VkImageLayout new_layout)
 {
     d3d12_command_list_transition_image_layout_with_global_memory_barrier(list, batch,
-            vk_image, vk_subresource,
+            resource, vk_subresource,
             src_stages, src_access, old_layout,
             dst_stages, dst_access, new_layout,
             0, 0);
@@ -10051,7 +10056,7 @@ static void d3d12_command_list_copy_image_transition_images(struct d3d12_command
         else
             old_layout = dst_resource->common_layout;
 
-        d3d12_command_list_transition_image_layout(list, batch, dst_resource->res.vk_image,
+        d3d12_command_list_transition_image_layout(list, batch, dst_resource,
                 &region->dstSubresource,
                 vk_availability_stages, VK_ACCESS_2_NONE,
                 old_layout,
@@ -10071,7 +10076,7 @@ static void d3d12_command_list_copy_image_transition_images(struct d3d12_command
         else
             old_layout = src_resource->common_layout;
 
-        d3d12_command_list_transition_image_layout(list, batch, src_resource->res.vk_image,
+        d3d12_command_list_transition_image_layout(list, batch, src_resource,
             &region->srcSubresource, vk_availability_stages,
             VK_ACCESS_2_NONE, old_layout,
             barrier.src.stages, barrier.src.access, barrier.src.layout);
@@ -10320,7 +10325,7 @@ static void d3d12_command_list_copy_image(struct d3d12_command_list *list,
         else
             new_layout = dst_resource->common_layout;
 
-        d3d12_command_list_transition_image_layout(list, batch, dst_resource->res.vk_image,
+        d3d12_command_list_transition_image_layout(list, batch, dst_resource,
                 &region->dstSubresource, barrier.dst.stages,
                 barrier.dst.access, barrier.dst.layout,
                 outside_vk_stages, outside_vk_access,
@@ -10336,7 +10341,7 @@ static void d3d12_command_list_copy_image(struct d3d12_command_list *list,
         else
             new_layout = src_resource->common_layout;
 
-        d3d12_command_list_transition_image_layout(list, batch, src_resource->res.vk_image,
+        d3d12_command_list_transition_image_layout(list, batch, src_resource,
             &region->srcSubresource, barrier.src.stages,
             VK_ACCESS_2_NONE, barrier.src.layout,
             outside_vk_stages, outside_vk_access,
@@ -10721,7 +10726,7 @@ static void d3d12_command_list_before_copy_texture_region(struct d3d12_command_l
             list->transfer_batch.tracked_copy_buffer_count = 0;
 
             d3d12_command_list_transition_image_layout_with_global_memory_barrier(list, batch,
-                    src_resource->res.vk_image,
+                    src_resource,
                     &info->copy.buffer_image.imageSubresource, VK_PIPELINE_STAGE_2_COPY_BIT, VK_ACCESS_2_NONE,
                     src_resource->common_layout, VK_PIPELINE_STAGE_2_COPY_BIT, VK_ACCESS_2_TRANSFER_READ_BIT,
                     info->src_layout, global_transfer_access, global_transfer_access);
@@ -10733,7 +10738,7 @@ static void d3d12_command_list_before_copy_texture_region(struct d3d12_command_l
 
         if (!unified || info->writes_full_subresource)
         {
-            d3d12_command_list_transition_image_layout(list, batch, dst_resource->res.vk_image,
+            d3d12_command_list_transition_image_layout(list, batch, dst_resource,
                     &info->copy.buffer_image.imageSubresource, VK_PIPELINE_STAGE_2_COPY_BIT, VK_ACCESS_2_NONE,
                     info->writes_full_subresource ? VK_IMAGE_LAYOUT_UNDEFINED : dst_resource->common_layout,
                     VK_PIPELINE_STAGE_2_COPY_BIT, VK_ACCESS_2_TRANSFER_WRITE_BIT, info->dst_layout);
@@ -10790,7 +10795,7 @@ static void d3d12_command_list_copy_texture_region(struct d3d12_command_list *li
         if (!unified)
         {
             d3d12_command_list_transition_image_layout(list, batch,
-                    src_resource->res.vk_image,
+                    src_resource,
                     &info->copy.buffer_image.imageSubresource, VK_PIPELINE_STAGE_2_COPY_BIT, VK_ACCESS_2_NONE,
                     info->src_layout, VK_PIPELINE_STAGE_2_COPY_BIT, VK_ACCESS_2_NONE, src_resource->common_layout);
         }
@@ -10816,7 +10821,7 @@ static void d3d12_command_list_copy_texture_region(struct d3d12_command_list *li
 
         if (!unified)
         {
-            d3d12_command_list_transition_image_layout(list, batch, dst_resource->res.vk_image,
+            d3d12_command_list_transition_image_layout(list, batch, dst_resource,
                     &info->copy.buffer_image.imageSubresource, VK_PIPELINE_STAGE_2_COPY_BIT,
                     VK_ACCESS_2_TRANSFER_WRITE_BIT, info->dst_layout, VK_PIPELINE_STAGE_2_COPY_BIT,
                     VK_ACCESS_2_NONE, dst_resource->common_layout);

diff --git a/vkd3d_build.h.in b/vkd3d_build.h.in
--- a/vkd3d_build.h.in
+++ b/vkd3d_build.h.in
@@ -1,3 +1,3 @@
 #include <stdint.h>
 
-static const uint64_t vkd3d_build = 0x@VCS_TAG@;
+static const uint64_t vkd3d_build = 0x628afa6f9cfece4;
diff --git a/vkd3d_version.h.in b/vkd3d_version.h.in
--- a/vkd3d_version.h.in
+++ b/vkd3d_version.h.in
@@ -1 +1 @@
-static const char vkd3d_version[] = "@VCS_TAG@";
+static const char vkd3d_version[] = "651f17762e439fe-flightdeck-r2";
