diff --git i/libs/vkd3d/device_vkd3d_ext.c w/libs/vkd3d/device_vkd3d_ext.c
index 94818b9a..c0057e17 100644
--- i/libs/vkd3d/device_vkd3d_ext.c
+++ w/libs/vkd3d/device_vkd3d_ext.c
@@ -1179,6 +1179,70 @@ CONST_VTBL struct ID3D12DXVKInteropDevice3Vtbl d3d12_dxvk_interop_device_vtbl =
     d3d12_dxvk_interop_device_GetVulkanHeapInfo,
 };
 
+void d3d12_device_notify_vk_swapchain_creation(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
+{
+    if (!device->vk_info.NV_low_latency2)
+        return;
+
+    spinlock_acquire(&device->low_latency_swapchain_spinlock);
+
+    if (device->swapchain_info.low_latency_swapchain && device->swapchain_info.low_latency_swapchain != chain)
+    {
+        /* Hard evidence of multiple swapchains being in flight. */
+        dxgi_vk_swap_chain_set_latency_sleep_mode(chain, false, false, 0);
+        dxgi_vk_swap_chain_decref(device->swapchain_info.low_latency_swapchain);
+        WARN("Multiple swapchains are in-flight. LL2 will be disabled until the situation stabilizes.\n");
+        device->swapchain_info.low_latency_swapchain = NULL;
+    }
+
+    spinlock_release(&device->low_latency_swapchain_spinlock);
+}
+
+void d3d12_device_register_low_latency_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
+{
+    if (!device->vk_info.NV_low_latency2)
+        return;
+
+    spinlock_acquire(&device->low_latency_swapchain_spinlock);
+
+    device->swapchain_info.vk_swapchain_count++;
+
+    if (device->swapchain_info.vk_swapchain_count == 1 && !device->swapchain_info.low_latency_swapchain)
+    {
+        dxgi_vk_swap_chain_incref(chain);
+        device->swapchain_info.low_latency_swapchain = chain;
+        dxgi_vk_swap_chain_set_latency_sleep_mode(chain, device->swapchain_info.mode,
+                device->swapchain_info.boost, device->swapchain_info.minimum_us);
+    }
+
+    /* Defer the demotion of the existing low-latency swapchain.
+     * Supposedly, there are applications that create a second swapchain, but never actually present anything to it
+     * before destroying it.
+     * It will never create a Vulkan swapchain and will not conflict with LL2.
+     * We can defer the demotion until we have evidence later. */
+
+    spinlock_release(&device->low_latency_swapchain_spinlock);
+}
+
+void d3d12_device_remove_low_latency_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
+{
+    if (!device->vk_info.NV_low_latency2)
+        return;
+
+    spinlock_acquire(&device->low_latency_swapchain_spinlock);
+
+    assert(device->swapchain_info.vk_swapchain_count);
+    device->swapchain_info.vk_swapchain_count--;
+
+    if (device->swapchain_info.low_latency_swapchain == chain)
+    {
+        dxgi_vk_swap_chain_decref(chain);
+        device->swapchain_info.low_latency_swapchain = NULL;
+    }
+
+    spinlock_release(&device->low_latency_swapchain_spinlock);
+}
+
 static inline struct d3d12_device *d3d12_device_from_ID3DLowLatencyDevice(d3d_low_latency_device_iface *iface)
 {
     return CONTAINING_RECORD(iface, struct d3d12_device, ID3DLowLatencyDevice_iface);
diff --git i/libs/vkd3d/swapchain.c w/libs/vkd3d/swapchain.c
index 5f2e9355..7140f6c8 100644
--- i/libs/vkd3d/swapchain.c
+++ w/libs/vkd3d/swapchain.c
@@ -623,7 +623,7 @@ static ULONG STDMETHODCALLTYPE dxgi_vk_swap_chain_AddRef(IDXGIVkSwapChain2 *ifac
     {
         dxgi_vk_swap_chain_incref(chain);
         ID3D12CommandQueue_AddRef(&chain->queue->ID3D12CommandQueue_iface);
-        d3d12_device_register_swapchain(chain->queue->device, chain);
+        d3d12_device_register_low_latency_swapchain(chain->queue->device, chain);
     }
 
     return refcount;
@@ -644,10 +644,7 @@ static ULONG STDMETHODCALLTYPE dxgi_vk_swap_chain_Release(IDXGIVkSwapChain2 *ifa
         /* Calling this from the submission thread will result in a deadlock, so
          * drain the swapchain queue now. */
         dxgi_vk_swap_chain_drain_queue(chain);
-
-        if (device->vk_info.NV_low_latency2)
-            d3d12_device_remove_swapchain(device, chain);
-
+        d3d12_device_remove_low_latency_swapchain(device, chain);
         dxgi_vk_swap_chain_decref(chain);
         ID3D12CommandQueue_Release(&queue->ID3D12CommandQueue_iface);
     }
@@ -1714,13 +1711,7 @@ static void dxgi_vk_swap_chain_destroy_swapchain_in_present_task(struct dxgi_vk_
     chain->present.current_backbuffer_index = UINT32_MAX;
 
     if (chain->queue->device->vk_info.NV_low_latency2)
-    {
-        spinlock_acquire(&chain->queue->device->low_latency_swapchain_spinlock);
-        chain->queue->device->swapchain_info.vk_swapchain_count--;
-        spinlock_release(&chain->queue->device->low_latency_swapchain_spinlock);
-
         pthread_mutex_unlock(&chain->present.low_latency_swapchain_lock);
-    }
 }
 
 static VkColorSpaceKHR convert_color_space(DXGI_COLOR_SPACE_TYPE dxgi_color_space)
@@ -2100,6 +2091,10 @@ static void dxgi_vk_swap_chain_recreate_swapchain_in_present_task(struct dxgi_vk
     if (chain->present.is_surface_lost)
         return;
 
+    /* Hard evidence that we're going ahead with swapchain creation, demote any existing LL2 chain
+     * that is not ourselves. */
+    d3d12_device_notify_vk_swapchain_creation(chain->queue->device, chain);
+
     /* If we fail to query formats we are hosed, treat it as a SURFACE_LOST scenario. */
     pthread_mutex_lock(&chain->properties.lock);
     /* This is only called on an event where we have to recreate the swapchain,
@@ -2283,18 +2278,6 @@ static void dxgi_vk_swap_chain_recreate_swapchain_in_present_task(struct dxgi_vk
     /* If low latency is supported restore the current low latency state now */
     if (chain->queue->device->vk_info.NV_low_latency2)
     {
-        struct d3d12_device *device = chain->queue->device;
-
-        spinlock_acquire(&device->low_latency_swapchain_spinlock);
-        device->swapchain_info.vk_swapchain_count++;
-
-        if (device->swapchain_info.vk_swapchain_count > 1 && device->swapchain_info.low_latency_swapchain)
-        {
-            dxgi_vk_swap_chain_decref(device->swapchain_info.low_latency_swapchain);
-            device->swapchain_info.low_latency_swapchain = NULL;
-        }
-        spinlock_release(&device->low_latency_swapchain_spinlock);
-
         dxgi_vk_swap_chain_set_low_latency_state(chain, &chain->present.low_latency_state);
         pthread_mutex_unlock(&chain->present.low_latency_swapchain_lock);
     }
@@ -3958,8 +3941,7 @@ static HRESULT STDMETHODCALLTYPE dxgi_vk_swap_chain_factory_CreateSwapChain(IDXG
         return hr;
     }
 
-    if (chain->queue->device->vk_info.NV_low_latency2)
-        d3d12_device_register_swapchain(chain->queue->device, chain);
+    d3d12_device_register_low_latency_swapchain(chain->queue->device, chain);
 
     *ppSwapchain = (IDXGIVkSwapChain*)&chain->IDXGIVkSwapChain_iface;
     return S_OK;
diff --git i/libs/vkd3d/vkd3d_private.h w/libs/vkd3d/vkd3d_private.h
index 5f94cfcf..b485d75c 100644
--- i/libs/vkd3d/vkd3d_private.h
+++ w/libs/vkd3d/vkd3d_private.h
@@ -6189,33 +6189,9 @@ static inline bool d3d12_device_prefers_render_pass_resolves(const struct d3d12_
             device->workarounds.tiler_suspend_resume_relax_load_store_op;
 }
 
-static inline void d3d12_device_register_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
-{
-    spinlock_acquire(&device->low_latency_swapchain_spinlock);
-
-    if (!device->swapchain_info.low_latency_swapchain)
-    {
-        dxgi_vk_swap_chain_incref(chain);
-        device->swapchain_info.low_latency_swapchain = chain;
-        dxgi_vk_swap_chain_set_latency_sleep_mode(chain, device->swapchain_info.mode,
-                device->swapchain_info.boost, device->swapchain_info.minimum_us);
-    }
-
-    spinlock_release(&device->low_latency_swapchain_spinlock);
-}
-
-static inline void d3d12_device_remove_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain)
-{
-    spinlock_acquire(&device->low_latency_swapchain_spinlock);
-
-    if (device->swapchain_info.low_latency_swapchain == chain)
-    {
-        dxgi_vk_swap_chain_decref(chain);
-        device->swapchain_info.low_latency_swapchain = NULL;
-    }
-
-    spinlock_release(&device->low_latency_swapchain_spinlock);
-}
+void d3d12_device_register_low_latency_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain);
+void d3d12_device_remove_low_latency_swapchain(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain);
+void d3d12_device_notify_vk_swapchain_creation(struct d3d12_device *device, struct dxgi_vk_swap_chain *chain);
 
 /* ID3DBlob */
 struct d3d_blob
